From 53fc121e5967303633524819bef39af0b4653d83 Mon Sep 17 00:00:00 2001
From: Weiguang Kong <weiguang.kong@nxp.com>
Date: Thu, 14 Sep 2017 13:31:22 +0800
Subject: [PATCH 2538/5242] MLK-16468-1: include: uapi: add multi-codec
 support for hifi4

commit  006c237759467ee62a6bef7e5ad8ab1e34ff3d82 from
https://source.codeaurora.org/external/imx/linux-imx.git

update the mxc_hifi4.h header file to support multi-codec
decoding or encoding together for hifi4 dsp.

Signed-off-by: Weiguang Kong <weiguang.kong@nxp.com>
Reviewed-by: Mihai Serban <mihai.serban@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/uapi/linux/mxc_hifi4.h |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/uapi/linux/mxc_hifi4.h b/include/uapi/linux/mxc_hifi4.h
index f8db598..4da1700 100644
--- a/include/uapi/linux/mxc_hifi4.h
+++ b/include/uapi/linux/mxc_hifi4.h
@@ -33,7 +33,7 @@
 #define HIFI4_DECODE_ONE_FRAME		_IOW(HIFI4_IOC_MAGIC, 4, unsigned int)
 #define HIFI4_UNLOAD_CODEC		_IOW(HIFI4_IOC_MAGIC, 5, unsigned int)
 #define HIFI4_GET_PCM_PROP		_IOW(HIFI4_IOC_MAGIC, 6, unsigned int)
-#define HIFI4_SET_CONFIG               _IOW(HIFI4_IOC_MAGIC, 7, unsigned int)
+#define HIFI4_SET_CONFIG		_IOW(HIFI4_IOC_MAGIC, 7, unsigned int)
 
 #define CODEC_MP3_DEC		1
 #define CODEC_AAC_DEC		2
@@ -54,6 +54,7 @@ struct decode_info {
 	int   out_buf_off;
 	unsigned int cycles;
 	unsigned int input_over;
+	unsigned int process_id;
 };
 
 struct prop_info {
@@ -69,11 +70,14 @@ struct prop_info {
 	int   aac_samplerate;
 	int   sbr_type;
 	int   mpeg_surr_present;
+
+	unsigned int process_id;
 };
 
 struct binary_info {
 	int type;
 	char *file;
+	unsigned int process_id;
 };
 
 struct prop_config {
@@ -81,6 +85,7 @@ struct prop_config {
 	int   cmd;       /* command value */
 	int   val;       /* parameter value */
 	int   ret;       /* executed status of function */
+	unsigned int process_id;
 };
 
 #endif/* __MXC_HIFI4_UAPI_H__ */
-- 
1.7.9.5

